File: z-clients/javascript/src/models/adminAuthTokenRotateOut.ts
// this file is @generated

export interface AdminAuthTokenRotateOut {
    id: string;
    token: string;
    created: Date;
    updated: Date;
}

export const AdminAuthTokenRotateOutSerializer = {
    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _fromJsonObject(object: any): AdminAuthTokenRotateOut {
        return {
            id: object['id'],
            token: object['token'],
            created: new Date(object['created']),
            updated: new Date(object['updated']),
        };
    },

    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _toJsonObject(self: AdminAuthTokenRotateOut): any {
        return {
            'id': self.id,
            'token': self.token,
            'created': self.created.getTime(),
            'updated': self.updated.getTime(),
        };
    }
}

File: z-clients/rust/src/models/admin_auth_token_rotate_out.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct AdminAuthTokenRotateOut {
    pub id: String,

    pub token: String,

    pub created: u64,

    pub updated: u64,
}

impl AdminAuthTokenRotateOut {
    pub fn new(id: String, token: String, created: u64, updated: u64) -> Self {
        Self {
            id,
            token,
            created,
            updated,
        }
    }
}
